sema4ai/src/sema4ai_code/refresh_agent_spec_helper.py

```diff
@@ -0,0 +1,128 @@
+from pathlib import Path
+from typing import Optional, TypedDict
+
+from ruamel.yaml import YAML
+
+
+class ActionPackage(TypedDict, total=False):
+    name: str
+    organization: str
+    type: str
+    version: str
+    whitelist: str
+    path: str
+
+
+class WhitelistEntry(TypedDict):
+    name: str
+    version: Optional[str]
+    organization: str
+    path: str
+    whitelist: Optional[str]
+
+
+def load_yaml(filepath: Path, yaml: YAML) -> None:
```

Comment:
I'd just inline this function (if you go for using `list_actions_from_agent` it'll only be used once... if you want to keep it, the return type must be fixed).